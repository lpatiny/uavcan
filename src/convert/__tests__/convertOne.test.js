'use strict';

const fs = require('fs');
const { join } = require('path');

const convertOne = require('../convertOne');

describe('convert', () => {
  it('simple', () => {
    const source = join(
      __dirname,
      '../../../dsdl/equipment/ahrs/1002.MagneticFieldStrength2.uavcan'
    );

    let content = fs.readFileSync(source, 'utf8');

    let result = convertOne(content, 'MagneticFieldStrength2');

    // console.log(JSON.stringify(result, undefined, 2));

    expect(result).toMatchSnapshot();
  });

  it('requestresponse', () => {
    const source = join(
      __dirname,
      '../../../dsdl/protocol/param/11.GetSet.uavcan'
    );

    let content = fs.readFileSync(source, 'utf8');

    let result = convertOne(content, 'GetSet');

    console.log(JSON.stringify(result, undefined, 2));

    expect(result).toMatchSnapshot();
  });
});
