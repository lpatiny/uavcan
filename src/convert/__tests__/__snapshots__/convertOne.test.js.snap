// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convert requestresponse 1`] = `
Object {
  "description": "Get or set a parameter by name or by index.
Note that access by index should only be used to retrieve the list of parameters; it is highly
discouraged to use it for anything else, because persistent ordering is not guaranteed.
Index of the parameter starting from 0; ignored if name is nonempty.
Use index only to retrieve the list of parameters.
Parameter ordering must be well defined (e.g. alphabetical, or any other stable ordering),
in order for the index access to work.
If set - parameter will be assigned this value, then the new value will be returned.
If not set - current parameter value will be returned.
Refer to the definition of Value for details.
Name of the parameter; always preferred over index if nonempty.
Actual parameter value.
For set requests, it should contain the actual parameter value after the set request was
executed. The objective is to let the client know if the value could not be updated, e.g.
due to its range violation, etc.
Empty value (and/or empty name) indicates that there is no such parameter.
Empty name (and/or empty value) in response indicates that there is no such parameter.",
  "request": Object {
    "statics": Array [],
    "variables": Array [
      Object {
        "bits": 13,
        "description": "",
        "kind": "int",
        "name": "index",
        "type": "var",
        "unsigned": true,
      },
      Object {
        "description": "",
        "kind": "Value",
        "name": "value",
        "type": "object",
      },
      Object {
        "description": "",
        "kind": Object {
          "bits": 8,
          "kind": "int",
          "type": "var",
          "unsigned": true,
        },
        "length": 92,
        "name": "name",
        "range": true,
        "type": "array",
      },
    ],
  },
  "response": Object {
    "statics": Array [],
    "variables": Array [
      Object {
        "bits": 5,
        "description": "",
        "kind": "void",
        "type": "var",
        "unsigned": false,
      },
      Object {
        "description": "",
        "kind": "Value",
        "name": "value",
        "type": "object",
      },
      Object {
        "bits": 5,
        "description": "",
        "kind": "void",
        "type": "var",
        "unsigned": false,
      },
      Object {
        "description": "Optional",
        "kind": "Value",
        "name": "defaultValue",
        "type": "object",
      },
      Object {
        "bits": 6,
        "description": "",
        "kind": "void",
        "type": "var",
        "unsigned": false,
      },
      Object {
        "description": "Optional, not applicable for bool/string",
        "kind": "NumericValue",
        "name": "maxValue",
        "type": "object",
      },
      Object {
        "bits": 6,
        "description": "",
        "kind": "void",
        "type": "var",
        "unsigned": false,
      },
      Object {
        "description": "Optional, not applicable for bool/string",
        "kind": "NumericValue",
        "name": "minValue",
        "type": "object",
      },
      Object {
        "description": "",
        "kind": Object {
          "bits": 8,
          "kind": "int",
          "type": "var",
          "unsigned": true,
        },
        "length": 92,
        "name": "name",
        "range": true,
        "type": "array",
      },
    ],
  },
  "type": "service",
}
`;

exports[`convert simple 1`] = `
Object {
  "description": "Magnetic field readings, in Gauss, in body frame.
SI units are avoided because of float16 range limitations.",
  "message": Object {
    "statics": Array [],
    "variables": Array [
      Object {
        "bits": 8,
        "description": "",
        "kind": "int",
        "name": "sensorId",
        "type": "var",
        "unsigned": true,
      },
      Object {
        "description": "",
        "kind": Object {
          "bits": 16,
          "kind": "float",
          "type": "var",
          "unsigned": false,
        },
        "length": 3,
        "name": "magneticFieldGa",
        "type": "array",
      },
      Object {
        "description": "",
        "kind": Object {
          "bits": 16,
          "kind": "float",
          "type": "var",
          "unsigned": false,
        },
        "length": 9,
        "name": "magneticFieldCovariance",
        "range": true,
        "type": "array",
      },
    ],
  },
  "type": "message",
}
`;
